flocker/acceptance/test_api.py

```diff
@@ -4,112 +4,44 @@
 Tests for the control service REST API.
 """
 
+from os import environ
 import socket
 
-from signal import SIGINT
-from os import kill
 from uuid import uuid4
 from json import dumps, loads
 
+from twisted.internet.defer import succeed
 from twisted.trial.unittest import TestCase
+from unittest import SkipTest
 from treq import get, post, content, delete, json_content
-from characteristic import attributes
+from pyrsistent import PRecord, field
 
-from .testtools import get_nodes, run_SSH
-from ..testtools import loop_until, random_name
+from ..testtools import loop_until
 
 from ..control.httpapi import REST_API_PORT
 
 
-def wait_for_api(hostname):
+class Node(PRecord):
     """
-    Wait until REST API is available.
-
-    :param str hostname: The host where the control service is
-         running.
-
-    :return Deferred: Fires when REST API is available.
-    """
-    def api_available():
-        try:
-            s = socket.socket()
-            s.connect((hostname, REST_API_PORT))
-            return True
-        except socket.error:
-            return False
-    return loop_until(api_available)
-
-
-@attributes(['address', 'agents'])
-class Node(object):
-    """
-    A record of a cluster node and its agents.
+    A record of a cluster node.
 
     :ivar bytes address: The IPv4 address of the node.
-    :ivar list agents: A list of ``RemoteService`` instances for the
-        convergence agent processes that have been started on this node.
-    """
-
-
-@attributes(['address', 'process'])
-class RemoteService(object):
-    """
-    A record of a background SSH process and the node that it's running on.
-
-    :ivar bytes address: The IPv4 address on which the service is running.
-    :ivar Subprocess.Popen process: The running ``SSH`` process that is running
-        the remote process.
-    """
-
-
-def close(process):
     """
-    Kill a process.
-
-    :param subprocess.Popen process: The process to be killed.
-    """
-    process.stdin.close()
-    kill(process.pid, SIGINT)
-
+    address = field(type=bytes)
```

Comment:
:question: Would it perhaps be better to use an `IPv4Address` instead of `bytes`.